packages/contracts/contracts/governance/UpgradeGovernor.sol

```diff
@@ -120,6 +120,41 @@ contract UpgradeGovernor is
         return super.proposalThreshold();
     }
 
+    /**
+     * @notice Function to queue a proposal to the timelock.
+     *         Added protocol for using custom time-lock zero delay for urgent situations.
+     */
+    function queue(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) public virtual override returns (uint256) {
+        uint256 proposalId = hashProposal(targets, values, calldatas, descriptionHash);
+
+        require(state(proposalId) == ProposalState.Succeeded, "Governor: proposal not successful");
+
+        uint256 delay = _timelock.getMinDelay();
+        // Protocol for reflecting urgent decisions on proposals.
+        // A zero delay is applied if the first element of the argument satisfies conditions.
+        if (targets[0] == address(0) && values[0] == 0 && calldatas[0].length == 0) {
+            delay = 0;
```

Comment:
Is the delay always 0 in urgent situations?

Reply:
I heard that we made decision to use `values[0]` as the value of `delay` in day unit, isn't it?

Reply:
I heard that we made decision to use `values[0]` as the value of I heard that we made decision to use values[0] as the value of delay in day unit, isn't it? in day unit, isn't it?

Reply:
I want to simplify the rule for emergencies to take only 0 time-locks.
This can be done through the use of calldata, but I expect that simplifying the exceptional behavior will make it clearer.